src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write code here
+const headers = document.querySelectorAll('thead th');
+const tbody = document.querySelector('tbody');
+
+const getCellValue = (row, index) => {
+  const cellText = row.children[index].innerText;
+
+  return cellText.startsWith('$')
+    ? parseFloat(cellText.replace(/[$,]/g, ''))
+    : cellText;
+};
+
+headers.forEach((header, index) => {
+  header.addEventListener('click', () => {
+    const rows = Array.from(tbody.querySelectorAll('tr'));
+
+    rows.sort((rowA, rowB) => {
+      const cellA = getCellValue(rowA, index);
+      const cellB = getCellValue(rowB, index);
+
+      return typeof cellA === 'number' && typeof cellB === 'number'
+        ? cellA - cellB
+        : cellA.localeCompare(cellB);
```

Comment:
The current sorting logic assumes that all values in a column are either numbers or strings. If a column contains mixed data types (e.g., numbers and strings), this logic might not work as expected. Consider adding a check to handle such cases or ensure that columns are consistently typed.